### PR TITLE
Add universaltest package

### PR DESCRIPTION
**Pull request by bourtemb**
Add recipe for UniversalTest Tango-Controls device server

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bourtemb** [comment]
@beenje , could you please confirm here that you are willing to be listed as maintainer of this universaltest package?
Many thanks in advance.

**Comment by beenje** [comment]
I'm willing to be a maintainer!

**Comment by beenje** [comment]
I see it's failing on windows with cpptango 9.3. Is is supposed to pass? We could build for 9.4 and 9.5 only otherwise.

**Comment by bourtemb** [comment]
> I see it's failing on windows with cpptango 9.3. Is is supposed to pass? We could build for 9.4 and 9.5 only otherwise.

I based my recipe on the one used for the Starter. I guess it is working ok for the Starter? I just changed the DCMAKE_BUILD_TYPE to Debug instead of Release in bld.bat.

**Comment by bourtemb** [comment]
When looking at the errors on Windows, it looks like we are encountering this issue when compiling on Windows with cpptango 9.3
: https://gitlab.com/tango-controls/cppTango/-/issues/680

So we should probably not try to compile UniversalTest with C++17 when using cppTango 9.3.
Do you know how we could do that?

**Comment by bourtemb** [comment]
> > I see it's failing on windows with cpptango 9.3. Is is supposed to pass? We could build for 9.4 and 9.5 only otherwise.
> 
> I based my recipe on the one used for the Starter. I guess it is working ok for the Starter? I just changed the DCMAKE_BUILD_TYPE to Debug instead of Release in bld.bat.

Well, I actually based my recipe on the ones from tango-starter and tango-test.
I just noticed that tango-starter is not compiling on Windows with cppTango 9.3.
So let's do the same here.

**Comment by bourtemb** [comment]
@conda-forge/help-c-cpp this PR is ready for review!

File: recipes/universaltest/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+mkdir build
+cd build
+cmake ${CMAKE_ARGS} \
+      -DCMAKE_BUILD_TYPE=Debug \
```
Comment (xhochy) [inline recipes/universaltest/build.sh:4]:
```suggestion
      -DCMAKE_BUILD_TYPE=Release \
```

File: recipes/universaltest/bld.bat [hunk 2]
```diff
@@ -0,0 +1,17 @@
+setlocal EnableDelayedExpansion
+
+mkdir build
+cd build
+
+cmake -G "NMake Makefiles" ^
+      -DCMAKE_BUILD_TYPE=Debug ^
```
Comment (xhochy) [inline recipes/universaltest/bld.bat:7]:
```suggestion
      -DCMAKE_BUILD_TYPE=Release ^
```

**Review by xhochy — CHANGES_REQUESTED** [review]
What's the reason you aren't build for `Release` here?

**Comment by bourtemb** [comment]
> What's the reason you aren't build for `Release` here?

It's because UniversalTest will be used by people playing with Tango-Controls framework features and potentially detect some bugs in the framework.  
It's then easier to debug or to report bugs when it's compiled in Debug mode.  
Debugging capabilities have been privileged over performances here.

**Review by xhochy — APPROVED** [review]

**Pull request merged**